system-tests/e2e-testsupport/src/main/java/com/hotels/styx/servers/MockOriginServer.java

```diff
@@ -0,0 +1,218 @@
+
+public final class MockOriginServer {
+    private static final String FILES_ROOT = "__files";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MockOriginServer.class);
+    private static final int MAX_CONTENT_LENGTH = 256 * 1024;
+
+    private final String appId;
+    private final String originId;
+    private final int adminPort;
+
+    private int serverPort;
+    private final HttpServer adminServer;
+    private final HttpServer mockServer;
+
+    static {
+        System.setProperty("org.mortbay.log.class", "com.github.tomakehurst.wiremock.jetty.LoggerAdapter");
+    }
+
+    private ServiceManager services;
+
+    private MockOriginServer(String appId, String originId, int adminPort, int serverPort, HttpServer adminServer, HttpServer mockServer) {
+        this.appId = appId;
+        this.originId = originId;
+        this.adminPort = adminPort;
+        this.serverPort = serverPort;
+        this.adminServer = adminServer;
+        this.mockServer = mockServer;
+    }
+
+    private static WireMockApp wireMockApp() {
+        return new WireMockApp(
+                new NotImplementedRequestDelayControl(),
+                false,
+                stubMappings -> {
+
+                },
+                mappings -> {
+
+                },
+                false,
+                absent(),
+                emptyMap(),
+                null,
+                null
+        );
+    }
+
+    private static HttpServer createAdminServer(String originId, int adminPort, WireMockApp wireMockApp) {
+        return HttpServers.createHttpServer(
+                "mock-admin-" + originId,
+                new HttpConnectorConfig(adminPort),
+                adminHandler(wireMockApp));
+    }
+
+    public static MockOriginServer create(String appId, String originId, int adminPort, HttpConnectorConfig httpConfig) {
+        WireMockApp wireMockApp = wireMockApp();
+        HttpServer adminServer = createAdminServer(originId, adminPort, wireMockApp);
+        HttpServer mockServer = HttpServers.createHttpServer(
+                "mock-stub-" + originId,
+                httpConfig,
+                mockHandler(wireMockApp, new WireMockConfiguration()));
+        int serverPort = httpConfig.port();
+
+        return new MockOriginServer(appId, originId, adminPort, serverPort, adminServer, mockServer);
+    }
+
+    public static MockOriginServer create(String appId, String originId, int adminPort, HttpsConnectorConfig httpsConfig) {
+        WireMockApp wireMockApp = wireMockApp();
+        HttpServer adminServer = createAdminServer(originId, adminPort, wireMockApp);
+        HttpServer mockServer = HttpServers.createHttpsServer(
+                "mock-stub-" + originId,
+                httpsConfig,
+                mockHandler(wireMockApp, new WireMockConfiguration()));
+        int serverPort = httpsConfig.port();
+        System.out.println(format("Creating HTTPS MockOriginServer - %d, %d", adminPort, serverPort));
```

Comment:
If this is required it should use the `LOGGER`. `System.out.println` usually causes checkstyle violations - was this not the case here?

Also, if we log for HTTPS, we should log for HTTP too.

Reply:
I have removed. The print line statement t shouldn't have been there.